### PR TITLE
IBX 4348: "Create Product" article added in User Documentation

### PR DESCRIPTION
**Pull request by julitafalcondusza**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [IBX-4348](https://issues.ibexa.co/browse/IBX-4348)
| Versions      | 4.3, master

"Create Product" article added in User Documentation.

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date

### CODE REVIEW

File: docs/pim/create_product.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+1\. In the left panel, go to **Product catalog** -> **Products** and click **Create** button. 
+
+![Adding a Product](img/add_product.png "Adding a Product")
+
+2\. Choose product type.
+
+3\. Click **Add** button.
+
+4\. Fill in basic product information: name, product specification, description and, if needed, other specific attributes.
```
Comment (adamwojs) [inline docs/pim/create_product.md:33]: Sidenote: product structure is customizable on product type level

**Review by adamwojs — APPROVED** [review]

File: docs/pim/create_product.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+---
+description: Create new products or modify existing ones.
+---
+
+# Create product
+
+If your [user role](../permission_management/work_with_permissions.md) has the `Product/Edit` permission, you can modify product and add individual attributes or attribute groups. You can create new [products](products.md#products) to present your offer in the website, including product specification, and pricing.
```
[the remark this answers is not in the document]
Comment (julitafalcondusza) [reply docs/pim/create_product.md:7]: Break lines added

File: docs/pim/create_product.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+---
+description: Create new products or modify existing ones.
+---
+
+# Create product
+
+If your [user role](../permission_management/work_with_permissions.md) has the `Product/Edit` permission, you can modify product and add individual attributes or attribute groups. You can create new [products](products.md#products) to present your offer in the website, including product specification, and pricing. 
+Each product has a unique identifying product code (SKU). [Product consists of](products.md#product-completeness) content, attributes, assests, availability, price and translation.
+!!! note
```
[the remark this answers is not in the document]
Comment (julitafalcondusza) [reply docs/pim/create_product.md:9]: "Note" removed

File: docs/pim/create_product.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    To create a product you must first create [product type](products.md#product-types).
```
[the remark this answers is not in the document]
Comment (julitafalcondusza) [reply docs/pim/create_product.md:11]: "Note" removed

File: docs/pim/create_product.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+5\. Click **Create** button.
+
+![Creating a product](img/create_product.png "Creating a product")
+
+You can also [add image assets](work_with_product_assets.md) to products.
+
+!!! note
+
+    Product structure is customizable on product type level.
```
[the remark this answers is not in the document]
Comment (julitafalcondusza) [reply docs/pim/create_product.md:39]: "Note" removed

File: docs/pim/create_product.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+---
+description: Create new products or modify existing ones.
+---
+
+# Create product
```
[the remark this answers is not in the document]
Comment (julitafalcondusza) [reply docs/pim/create_product.md]: Ok, corrected

File: docs/pim/create_product.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+1\. In the left panel, go to **Product catalog** -> **Products**.
+
+2\. Click the **Edit** button next to a name of the product item that you want to modify.
+
+![Editing a product](img/edit_product.png "Editing a product")
+
+3\. Edit information and click **Update** button.
+
+## Work with products
```
[the remark this answers is not in the document]
Comment (julitafalcondusza) [reply docs/pim/create_product.md:53]: Ok, links added

File: mkdocs.yml [hunk 8]
```diff
@@ -56,6 +56,7 @@ nav:
               - Filters: personalization/filters.md
     - PIM:
         - Products: pim/products.md
+        - Create product: pim/create_product.md
```
[the remark this answers is not in the document]
Comment (julitafalcondusza) [reply mkdocs.yml:59]: Ok, corrected

File: docs/pim/create_product.md (hunk 1 above)
Comment (julitafalcondusza) [reply docs/pim/create_product.md:29]: Ok, description updated. 
As we discussed - let's wait for the update of 'Content Items' article, then we can add the link.

File: docs/pim/create_product.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    To create a product you must first create [product type](products.md#product-types).
+
+## Manually create product
+
+You can create product in several ways:
+
+- In the left panel, go to **Product catalog** -> **Products** and click **Create** button.
+- In the left panel, go to **Content** -> **Content structure** -> **Product catalog** and click **Create content** button.
+- In the left panel, go to **Content** -> **Dashboard** and click **Create content** button.
+- In the search bar type 'product' -> click **Product catalog** and click **Create** button.
```
[the remark this answers is not in the document]
Comment (julitafalcondusza) [reply docs/pim/create_product.md]: Ok, link to dev doc about bulk method added

File: docs/pim/create_product.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+You can edit existing products at any time:
+
+1\. In the left panel, go to **Product catalog** -> **Products**.
+
+2\. Click the **Edit** button next to a name of the product item that you want to modify.
+
+![Editing a product](img/edit_product.png "Editing a product")
+
+3\. Edit information and click **Update** button.
```
[the remark this answers is not in the document]
Comment (julitafalcondusza) [reply docs/pim/create_product.md]: Editing part merged with creating product procedure

File: docs/pim/create_product.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+![Adding a Product](img/add_product.png "Adding a Product")
+
+2\. Choose product type.
+
+3\. Click **Add** button.
+
+4\. Fill in basic product information: name, product specification, description and, if needed, other specific attributes. 
+
+5\. Click **Create** button.
```
[the remark this answers is not in the document]
Comment (julitafalcondusza) [reply docs/pim/create_product.md:31]: Ok, added

File: docs/pim/create_product.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+![Creating a product](img/create_product.png "Creating a product")
+
+You can also [add image assets](work_with_product_assets.md) to products.
```
[the remark this answers is not in the document]
Comment (julitafalcondusza) [reply docs/pim/create_product.md:35]: Ok, added

File: docs/pim/create_product.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+such as a name, SKU, description and images, attributes, assets, availability, prices, and translations.
+
+You can create product by using manual or bulk method.
+Bulk method can be used only at the developer level. See [Products](https://doc.ibexa.co/en/latest/pim/product_api/#products) for a technical guide on how to do this.
+
+## Manually create product
+
+You can create product in several ways:
+
+1. Click **Create** in the upper-right corner of the [Products](products.md) screen.
```
Comment (dabrt) [inline docs/pim/create_product.md:24]: I'd change this into a bulleted list - these not steps that need to be done in order, these are options to choose from.

File: docs/pim/create_product.md [hunk 14]
```diff
@@ -0,0 +1,56 @@
+---
+description: Create new products or modify existing ones.
+---
+
+# Create products
+
+[Products](products.md#products) are a specific kind of [Content items](../content_management/content_items.md#content-items) that you use 
+to present your offer in the website, including product specification, and pricing.
+
+Individual products are instances of [product types](create_product_types.md#create-product-types).
```
Comment (dabrt) [inline docs/pim/create_product.md:10]: I'd merge these three sentences into one paragraph. To do it, remove empty lines between them.

File: docs/pim/create_product.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+
+You can create product in several ways:
+
+1. Click **Create** in the upper-right corner of the [Products](products.md) screen. 
+2. While viewing a **Product Catalog** in the [Content Tree](../getting_started/discover_ui.md#content-tree), click **Create content** at the top of the screen.
+The new item appears in the tree.
+3. Click **Create content** in the upper-right corner of the [My dashboard](../getting_started/discover_ui.md) screen. In this case you have to choose Content Type and select where the content will be located.
+
+1\. If you are adding a new product use one of the methods mentioned above and continue with the procedure.
+If you want to edit existing product, click the **Edit** button next to a name of the product item that you want to modify and skip to step 3.
```
Comment (dabrt) [inline docs/pim/create_product.md:30]:
```suggestion
1\. If you are adding a new product, use one of the methods mentioned above and  and skip to step 3.

2\. If you are editing an existing product, click the **Edit** button next to a name of the product item that you want to modify and skip to step 4.
```

File: docs/pim/create_product.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+
+![Adding a Product](img/add_product.png "Adding a Product")
+
+![Editing a product](img/edit_product.png "Editing a product")
+
+2\. From their lists, select the language and the product type, and then click the **Add** button.
```
Comment (dabrt) [inline docs/pim/create_product.md:36]:
```suggestion
3\. From their lists, select the language and the product type, and then click the **Add** button.
```

File: docs/pim/create_product.md (hunk 14 above)
Comment (dabrt) [inline docs/pim/create_product.md:10]:
```suggestion
Individual products are instances of [Product Types](create_product_types.md#create-product-types).
```

File: docs/pim/create_product.md [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+
+3\. Fill in or edit content fields of the product, for example, name, specification and description.
+The [Fields](../content_management/content_model.md#fields-and-field-types) that you have to populate depend on how the Product Type is defined. 
+Fields marked with an asterisk (*) are required.
+
+4\. In the Attributes section, define the product's attributes, for example dimensions, resolution or capacity. Attributes depend on defined Product Type
```
Comment (dabrt) [inline docs/pim/create_product.md:42]:
```suggestion
5\. In the Attributes section, define the product's attributes, for example, dimensions, resolution or capacity. 
```

File: docs/pim/create_product.md [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+If you want to edit existing product, click the **Edit** button next to a name of the product item that you want to modify and skip to step 3.
+
+![Adding a Product](img/add_product.png "Adding a Product")
+
+![Editing a product](img/edit_product.png "Editing a product")
+
+2\. From their lists, select the language and the product type, and then click the **Add** button.
+
+3\. Fill in or edit content fields of the product, for example, name, specification and description.
+The [Fields](../content_management/content_model.md#fields-and-field-types) that you have to populate depend on how the Product Type is defined.
```
Comment (dabrt) [inline docs/pim/create_product.md:39]:
```suggestion
```

I'd remove this sentence from the step and merge it with a sentence in line 14 above, to create something like:

"To create a product, depending on how the Product Type is defined, you [may need to provide](products.md#product-completeness) certain pieces of information in their respective [Fields](../content_management/content_model.md#fields-and-field-types),"

File: docs/pim/create_product.md [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+1\. If you are adding a new product use one of the methods mentioned above and continue with the procedure.
+If you want to edit existing product, click the **Edit** button next to a name of the product item that you want to modify and skip to step 3.
+
+![Adding a Product](img/add_product.png "Adding a Product")
+
+![Editing a product](img/edit_product.png "Editing a product")
+
+2\. From their lists, select the language and the product type, and then click the **Add** button.
+
+3\. Fill in or edit content fields of the product, for example, name, specification and description.
```
Comment (dabrt) [inline docs/pim/create_product.md:38]:
```suggestion
4\. Fill in or edit content fields of the product, for example, name, specification and description.
```

File: docs/pim/create_product.md [hunk 20]
```diff
@@ -0,0 +1,56 @@
+---
+description: Create new products or modify existing ones.
+---
+
+# Create products
```
Comment (dabrt) [inline docs/pim/create_product.md:5]:
```suggestion
# Create and edit products
```

File: docs/pim/create_product.md [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+2\. From their lists, select the language and the product type, and then click the **Add** button.
+
+3\. Fill in or edit content fields of the product, for example, name, specification and description.
+The [Fields](../content_management/content_model.md#fields-and-field-types) that you have to populate depend on how the Product Type is defined. 
+Fields marked with an asterisk (*) are required.
+
+4\. In the Attributes section, define the product's attributes, for example dimensions, resolution or capacity. Attributes depend on defined Product Type
+and some of them can be required.
+
+5\. If you are adding new product, click **Create** button.
```
Comment (dabrt) [inline docs/pim/create_product.md:45]:
```suggestion
6\. If you are adding a new product, click the **Create** button.
```

File: docs/pim/create_product.md [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+If you are editing existing one, click **Update** button.
```
Comment (dabrt) [inline docs/pim/create_product.md:46]:
```suggestion
If you are editing an existing one, click the **Update** button.
```

File: docs/pim/create_product.md [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+
+![Creating a product](img/create_product.png "Creating a product")
+
+## Work with products
```
Comment (dabrt) [inline docs/pim/create_product.md:50]:
```suggestion
```

File: docs/pim/create_product.md [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+
+You can also [add image assets](work_with_product_assets.md) to the products, [create variants to the main product](work_with_product_variants.md), [define prices](manage_prices_and_stock.md) and [clasify products into different categories](work_with_product_categories.md).
```
Comment (dabrt) [inline docs/pim/create_product.md:52]:
```suggestion
After you create a product, you can [add image assets to a product](work_with_product_assets.md), [create variants to the main product](work_with_product_variants.md), [define product prices](manage_prices_and_stock.md), and [clasify products into different categories](work_with_product_categories.md).
```

File: docs/pim/create_product.md [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+
+Be aware that some functinalities may differ depending on the specifics of the installation.
```
Comment (dabrt) [inline docs/pim/create_product.md:54]:
```suggestion
!!! note

    Feature availability may differ depending on the specifics of your installation.
```

File: docs/persona_paths/manage_products.md [hunk 26]
```diff
@@ -11,6 +11,7 @@ your stock and customers can easily browse your catalogs.
 
 [[= cards([
     "pim/create_product_types",
+    "pim/create_product",
```
Comment (dabrt) [inline docs/persona_paths/manage_products.md:14]:
```suggestion
    "pim/create_edit_products",
```

Rename the `create_product.md` file, too

File: mkdocs.yml [hunk 27]
```diff
@@ -55,8 +55,9 @@ nav:
               - Scenarios: personalization/scenarios.md
               - Filters: personalization/filters.md
     - PIM:
-        - Products: pim/products.md
+        - Product Information Management: pim/products.md
         - Create product types: pim/create_product_types.md
+        - Create product: pim/create_product.md
```
Comment (dabrt) [inline mkdocs.yml:60]:
```suggestion
        - Create and edit products: pim/create_product.md
```

File: docs/pim/create_product.md (hunk 13 above)
Comment (julitafalcondusza) [reply docs/pim/create_product.md:24]: Ok, updated

File: docs/pim/create_product.md (hunk 14 above)
Comment (julitafalcondusza) [reply docs/pim/create_product.md]: Ok, updated

File: docs/pim/create_product.md (hunk 17 above)
Comment (julitafalcondusza) [reply docs/pim/create_product.md:43]: Ok, updated

File: docs/pim/create_edit_product.md [hunk 28]
```diff
@@ -0,0 +1,54 @@
...
+Individual products are instances of [Product Types](create_product_types.md#create-product-types).
+You can only create or modify products when your [user role](../permission_management/work_with_permissions.md) has the `Product/Edit` permission.
+
+To create a product, depending on how the Product Type is defined, you [may need to provide](products.md#product-completeness) certain pieces 
+of information in their respective [Fields](../content_management/content_model.md#fields-and-field-types).
+
+You can create product by using manual or bulk method.
+Bulk method can be used only at the developer level. See [Products](https://doc.ibexa.co/en/latest/pim/product_api/#products) for a technical guide on how to do this.
+
+## Manually create product
```
Comment (dabrt) [inline docs/pim/create_edit_product.md:18]: Do we need this heading at all? Would we miss anything if we deleted it? I start thinking that it is obsolete.

File: docs/pim/create_edit_product.md [hunk 29]
```diff
@@ -0,0 +1,54 @@
...
+Fields marked with an asterisk (*) are required.
+
+5\. In the Attributes section, define the product's attributes, for example, dimensions, resolution or capacity.
+
+6\. If you are adding a new product, click the **Create** button.
+If you are editing an existing one, click the **Update** button.
+
+![Creating a product](img/create_product.png "Creating a product")
+
+
```
Comment (dabrt) [inline docs/pim/create_edit_product.md:47]:
```suggestion
```

File: docs/pim/create_edit_product.md [hunk 30]
```diff
@@ -0,0 +1,54 @@
...
+You can create product in several ways:
+
+- Click **Create** in the upper-right corner of the [Products](products.md) screen. 
+- While viewing a **Product Catalog** in the [Content Tree](../getting_started/discover_ui.md#content-tree), click **Create content** at the top of the screen.
+The new item appears in the tree.
+- Click **Create content** in the upper-right corner of the [My dashboard](../getting_started/discover_ui.md) screen. In this case you have to choose Content Type and select where the content will be located.
+
+1\. If you are adding a new product, use one of the methods mentioned above and skip to step 3.
+
+![Adding a Product](img/add_product.png "Adding a Product")
```
Comment (dabrt) [inline docs/pim/create_edit_product.md:29]:
```suggestion
```

I think I would remove this screenshot. Instead, I would add the following image in step 3, with caption "Creating a new product".
obraz

File: docs/pim/create_edit_product.md [hunk 31]
```diff
@@ -0,0 +1,54 @@
...
+
+2\. If you are editing an existing product, click the **Edit** button next to a name of the product item that you want to modify and skip to step 4.
+
+![Editing a product](img/edit_product.png "Editing a product")
```
Comment (dabrt) [inline docs/pim/create_edit_product.md:33]:
```suggestion
![Products list with action buttons](img/edit_product.png "Products list with action buttons")
```

File: docs/pim/create_edit_product.md [hunk 32]
```diff
@@ -0,0 +1,54 @@
...
+
+4\. Fill in or edit content fields of the product, for example, name, specification and description.
+Fields marked with an asterisk (*) are required.
+
+5\. In the Attributes section, define the product's attributes, for example, dimensions, resolution or capacity.
+
+6\. If you are adding a new product, click the **Create** button.
+If you are editing an existing one, click the **Update** button.
+
+![Creating a product](img/create_product.png "Creating a product")
```
Comment (dabrt) [inline docs/pim/create_edit_product.md:45]:
```suggestion
![Editing product information](img/create_product.png "Editing product information")
```

File: docs/pim/create_edit_product.md [hunk 33]
```diff
@@ -0,0 +1,54 @@
...
+
+
+After you create a product, you can [add image assets to a product](work_with_product_assets.md), [create variants to the main product](work_with_product_variants.md), [define product prices](manage_prices_and_stock.md), and [clasify products into different categories](work_with_product_categories.md).
+
+!!! note
+
+    Feature availability may differ depending on the specifics of your installation.
+
+For more information, see [Products](https://doc.ibexa.co/en/latest/pim/products/).
```
Comment (dabrt) [inline docs/pim/create_edit_product.md:54]:
```suggestion
For in-depth information, see [Products](https://doc.ibexa.co/en/latest/pim/products/) in developer documentation.
```

**Review by dabrt — COMMENTED** [review]
Please see my comments. When you apply changes from comments, you can select "Add change to batch" and commit them all at once, instead of committing one by one.

Ping me if you have questions.

File: docs/pim/create_edit_product.md (hunk 28 above)
Comment (julitafalcondusza) [reply docs/pim/create_edit_product.md:18]: Agree

**Review by dabrt — APPROVED** [review]

**Pull request merged**